sh/binfmt.sh.in

```diff
@@ -60,14 +60,14 @@ else
 	# `/run/binfmt.d/foo.conf' supersedes `/usr/lib/binfmt.d/foo.conf'.
 	# `/run/binfmt.d/foo.conf' will always be read after `/etc/binfmt.d/bar.conf'
 	for d in ${binfmt_dirs} ; do
-		[ -d $d ] && for f in ${d}/*.conf ; do
+		[ -d "$d" ] && for f in "${d}"/*.conf ; do
 			case "${f##*/}" in
 				systemd.conf|systemd-*.conf) continue;;
 			esac
-			[ -e $f ] && binfmt_basenames="${binfmt_basenames}\n${f##*/}"
+			[ -e "$f" ] && binfmt_basenames="${binfmt_basenames}\\n${f##*/}"
 		done # for f in ${d}
 	done # for d in ${binfmt_dirs}
-	binfmt_basenames="$(printf "${binfmt_basenames}\n" | sort -u )"
+	binfmt_basenames="$(printf "%s" "${binfmt_basenames}\\n" | sort -u )"
```

Comment:
you need `%b` here, not `%s`.  otherwise you're breaking this code.

Reply:
Nice catch, thanks. Fixed.